server/routes/api.js

```diff
@@ -195,4 +247,40 @@ router.post('/reload', function(req, res) {
 	res.json(true);
 });
 
+router.post('/is-viewable', function(req, res){
+
+	const url = req.body.url;
+
+	fetch(url, {
+			method: 'head'
+		})
+		.then(function(response){
+			return response.headers.get('x-frame-options');
+		})
+		.then(function(xFrameHeader){
+
+			if(xFrameHeader === null){
+				res.json({
```

Comment:
Could this make use of the createResponse function that is defined at the top of the file?